tests/unit/test_lineage.py

```diff
@@ -175,3 +177,132 @@ def test_get_user_entity():
     assert _ObjectField.DOMAIN in user_entity
     assert user_entity[_ObjectField.DOMAIN] == _UserDomain.FEATURE_VIEW
     assert _ObjectField.CREATED_ON in user_entity
+
+
+def test_split_fully_qualified_name():
+    test_cases_valid = [
+        "my_database.public.sales_table",
+        '"my_database".public."sales_table"',
+        'database."schema"."object"',
+        '"database".schema."object"',
+        '"database"."schema".object',
+        '"data.base".schema.object',
+    ]
+    for each in test_cases_valid:
+        assert 3 == len(_DGQLQueryBuilder.split_fully_qualified_name(each))
+
+
+def test_is_valid_object_name():
+    fake_session = mock.create_autospec(Session, _session_id=123456)
+    fake_session._analyzer = mock.Mock()
+
+    test_cases_valid = [
+        "my_database.public.sales_table",
+        '"my_database".public."sales_table"',
+        'database."schema"."object"',
+        '"database".schema."object"',
+        '"database"."schema".object',
+        '"data.base".schema.object',
+    ]
+
+    test_cases_invalid = [
+        '"database.schema.object"',
+        '"my_database"."public.sales_table"',
+        '"databa"se".schema."object"',
+    ]
+
+    # Assert checks for valid cases
+    for case in test_cases_valid:
+        Lineage(fake_session)._check_valid_object_name(case)
+
+    # Assert checks for invalid cases
+    for case in test_cases_invalid:
+        with pytest.raises(ValueError) as exc:
+            Lineage(fake_session)._check_valid_object_name(case)
+        assert "Invalid object name:" in str(exc)
+
+
+def test_get_feature_view_name():
```

Comment:
can we test cases where the database/schema names are quoted and have special characters such as $ or dot?

Reply:
added